safehttp/plugins/csp/csp.go

```diff
@@ -92,77 +92,77 @@ type StrictCSPBuilder struct {
 	Hashes []string
 }
 
-// Build creates a Policy based on the specified options.
-func (s StrictCSPBuilder) Build() Policy {
-	return Policy{
-		serialize: func(nonce string) string {
-			var b strings.Builder
-
-			// object-src 'none'; script-src 'unsafe-inline' 'nonce-{random}'
-			b.WriteString("object-src 'none'; script-src 'unsafe-inline' 'nonce-")
-			b.WriteString(nonce)
-			b.WriteByte('\'')
-
-			if !s.NoStrictDynamic {
-				b.WriteString(" 'strict-dynamic' https: http:")
-			}
-
-			if s.UnsafeEval {
-				b.WriteString(" 'unsafe-eval'")
-			}
-
-			for _, h := range s.Hashes {
-				b.WriteString(" '")
-				b.WriteString(h)
-				b.WriteByte('\'')
-			}
-
-			b.WriteString("; base-uri ")
-			if s.BaseURI == "" {
-				b.WriteString("'none'")
-			} else {
-				b.WriteString(s.BaseURI)
-			}
-
-			if s.ReportURI != "" {
-				b.WriteString("; report-uri ")
-				b.WriteString(s.ReportURI)
-			}
-
-			return b.String()
-		},
+// Serialize serializes this policy for use in a Content-Security-Policy header
+// or in a Content-Security-Policy-Report-Only header. A nonce will be provided
+// to serialize which can be used in 'nonce-{random-nonce}' values in directives.
+func (s StrictPolicy) Serialize(nonce string) string {
+	var b strings.Builder
+
+	// object-src 'none'; script-src 'unsafe-inline' 'nonce-{random}'
+	b.WriteString("object-src 'none'; script-src 'unsafe-inline' 'nonce-")
+	b.WriteString(nonce)
+	b.WriteByte('\'')
+
+	if !s.NoStrictDynamic {
+		b.WriteString(" 'strict-dynamic' https: http:")
+	}
+
+	if s.UnsafeEval {
+		b.WriteString(" 'unsafe-eval'")
+	}
+
+	for _, h := range s.Hashes {
+		b.WriteString(" '")
+		b.WriteString(h)
+		b.WriteByte('\'')
+	}
+
+	b.WriteString("; base-uri ")
+	if s.BaseURI == "" {
+		b.WriteString("'none'")
+	} else {
+		b.WriteString(s.BaseURI)
 	}
+
+	if s.ReportURI != "" {
+		b.WriteString("; report-uri ")
+		b.WriteString(s.ReportURI)
+	}
+
+	return b.String()
+
 }
 
-// FramingPolicyBuilder can be used to create a new CSP policy with frame-ancestors
+// FramingPolicy can be used to create a new CSP policy with frame-ancestors
 // set to 'self'.
 //
 // TODO: allow relaxation on specific endpoints according to #77.
-type FramingPolicyBuilder struct {
+type FramingPolicy struct {
 	// ReportURI controls the report-uri directive. If ReportUri is empty, no report-uri
 	// directive will be set.
 	ReportURI string
 }
 
-// Build creates a Policy based on the specified options.
-func (f FramingPolicyBuilder) Build() Policy {
-	return Policy{
-		serialize: func(_ string) string {
-			var b strings.Builder
-			b.WriteString("frame-ancestors 'self'")
+// Serialize serializes this policy for use in a Content-Security-Policy header
+// or in a Content-Security-Policy-Report-Only header. A nonce will be provided
+// to serialize which can be used in 'nonce-{random-nonce}' values in directives.
+func (f FramingPolicy) Serialize(nonce string) string {
```

Comment:
FYI: None of the code has changed here. It's just indentation and wrapping around in a closure that got removed.